packages/multi-test/src/wasm.rs

```diff
@@ -389,28 +392,27 @@ where
         });
 
         // call reply if meaningful
-        if let Ok(r) = res {
+        if let Ok(mut r) = res {
             if matches!(reply_on, ReplyOn::Always | ReplyOn::Success) {
-                let mut orig = r.clone();
                 let reply = Reply {
                     id,
                     result: ContractResult::Ok(SubMsgExecutionResponse {
-                        events: r.events,
+                        events: r.events.clone(),
                         data: r.data,
                     }),
                 };
                 // do reply and combine it with the original response
-                let res2 = self._reply(api, router, storage, block, contract, reply)?;
-                // override data if set
-                if let Some(data) = res2.data {
-                    orig.data = Some(data);
-                }
+                let reply_res = self._reply(api, router, storage, block, contract, reply)?;
```

Comment:
Very good changes. We return the data from reply, not from the original sub message (which is only needed to fill into reply)